src/common/utils/validator.js

```diff
@@ -117,17 +117,23 @@ function isPrintable(value) {
   return !value.includes('\n');
 }
 
-// validday
 function validDay(d, m, y) {
   if (isNumeric(d)) {
-    return m >= 0 && m <= 12 && d > 0 && d <= daysInMonth(m, y);
+    return validMonth(m) && (d >= 1 && d <= daysInMonth(m, y));
   }
   return false;
 }
 
 function validMonth(m) {
   if (isNumeric(m)) {
-    return m >= 0 && m <= 12;
+    return m >= 1 && m <= 12;
+  }
+  return false;
+}
+
+function validYear(y) {
+  if (isNumeric(y)) {
+    return /^[1-9][0-9]{3}?$/gm.test(y); // Matches 4 digit years between 1000 and 9999
```

Comment:
Question: should regex be here or not?
* e.g. something like `1000 >= y && y <= 9999` does is what the comment is saying.

This is asked as the code is looking for a particular valid year number, using regex seems like overkill.

Reply:
Has been done, thanks for doing so.

Note, the `validYear` will only return `true` for number strings (`validYear("2000")` or `validYear("3010")`) rather than actual numbers (`validYear(2000)` or `year(3010)`).
* Is this intended behaviour or not?
* It should not error in this case, it can remain in if it is desired - if possible, give the reason why.

Part of code I am referring to:
```javascript
function validYear(y) {
  if (isNumeric(y)) {
    return (y.length === 4) && (y >= 1000 && y <= 9999);
  }
  return false;
}
```

Reasoning:
* This is because `Number` primitive in javascript (`2000` rather than `"2000"`) **don't** have a `length` property.
* `String` primitives e.g. `"bananas on toast"` and `"2000"` do have a `length` property.
* Therefore, even though `isNumeric(2000)` returns `true`, the below check will return `false`.
* If I am wrong, let me know.
* Since currently the app seems to send string number year in, so it is fine. Concern  is it may change input to a number year in the future though.